cmd/example-disk-mutation-hook-sidecar/diskimage.go

```diff
@@ -22,128 +22,77 @@
-	log.Log.Info("Disk mutation hook's OnDefineDomain callback method has been called")
-
-	domainXML := params.GetDomainXML()
-	vmiJSON := params.GetVmi()
+func OnDefineDomain(log *log.Logger, vmiJSON, domainXML []byte) (string, error) {
 	vmiSpec := vmSchema.VirtualMachineInstance{}
 	err := json.Unmarshal(vmiJSON, &vmiSpec)
 	if err != nil {
-		log.Log.Reason(err).Errorf("Failed to unmarshal given VMI spec: %s", vmiJSON)
-		panic(err)
+		return "", fmt.Errorf("Failed to unmarshal given VMI spec: %s %s", err, string(vmiJSON))
 	}
 
 	annotations := vmiSpec.GetAnnotations()
 	if _, found := annotations[bootDiskImageNameAnnotation]; !found {
-		log.Log.Info("Boot disk hook sidecar was requested, but no attributes provided. Returning original domain spec")
-		return &hooksV1alpha2.OnDefineDomainResult{
-			DomainXML: domainXML,
-		}, nil
+		log.Print("Boot disk hook sidecar was requested, but no attributes provided. Returning original domain spec")
+		return string(domainXML), nil
 	}
 
 	domainSpec := domainSchema.DomainSpec{}
 	err = xml.Unmarshal(domainXML, &domainSpec)
 	if err != nil {
-		log.Log.Reason(err).Errorf("Failed to unmarshal given domain spec: %s", domainXML)
-		panic(err)
+		return "", fmt.Errorf("Failed to unmarshal given domain spec: %s %s", err, string(domainXML))
 	}
 
 	// Get Boot disk
 	bootDiskLocation := domainSpec.Devices.Disks[0].Source.File
 	dir, diskName := filepath.Split(bootDiskLocation)
 	newDiskName := annotations[bootDiskImageNameAnnotation]
 	if newDiskName == diskName {
-		log.Log.Infof("Boot disk image name is already %q. Returning original domain spec", newDiskName)
-		return &hooksV1alpha2.OnDefineDomainResult{
-			DomainXML: domainXML,
-		}, nil
+		log.Printf("Boot disk image name is already %q. Returning original domain spec", newDiskName)
+		return string(domainXML), nil
 	}
 
 	bootDiskLocation = filepath.Join(dir, newDiskName)
 	domainSpec.Devices.Disks[0].Source.File = bootDiskLocation
 
 	newDomainXML, err := xml.Marshal(domainSpec)
 	if err != nil {
-		log.Log.Reason(err).Errorf("Failed to marshal updated domain spec: %+v", domainSpec)
-		panic(err)
+		return "", fmt.Errorf("Failed to marshal updated domain spec: %s %+v", err, domainSpec)
 	}
 
-	log.Log.Info("Successfully updated original domain spec with requested boot disk attribute")
-	return &hooksV1alpha2.OnDefineDomainResult{
-		DomainXML: newDomainXML,
-	}, nil
-}
-
-func (s v1alpha2Server) PreCloudInitIso(_ context.Context, params *hooksV1alpha2.PreCloudInitIsoParams) (*hooksV1alpha2.PreCloudInitIsoResult, error) {
-	return &hooksV1alpha2.PreCloudInitIsoResult{
-		CloudInitData: params.GetCloudInitData(),
-	}, nil
+	return string(newDomainXML), nil
 }
 
 func main() {
-	log.InitializeLogging("bootdisk-hook-sidecar")
-
-	var version string
-	pflag.StringVar(&version, "version", "", "hook version to use")
+	var vmiJSON, domainXML string
+	pflag.StringVar(&vmiJSON, "vmi", "", "VMI to change in JSON format")
```

Comment:
I think you've committed some temporary code here

Reply:
I don't know where? Moving to sidecar-shim container image means using pflag for the binary to receive the vmi (in json format) and the domain (in xml format). I might be missing it, so let me know